substrate/utils/frame/sub-du/src/main.rs

```diff
@@ -0,0 +1,380 @@
+	final_key
+}
+
+/// Using `state_getStorageSize` RPC call times out since querying storage size of a relatively
+/// large storage prefix takes a lot of time. This function is a workaround to get the size of the
+/// storage by using paginated `state_getKeysPaged` RPC call.
+///
+/// It is a modified implementation of `state_getStorageSize` RPC call, with some major differences:
+///
+/// - uses paginated reading of keys.
+/// - only reads once for the value size, instead of reading it for all keys.
+///
+/// The latter might not be entirely accurate, since not all values might have the same size (e.g.
+/// when storage value contains `Option<T>` or unbounded data types). It is a trade-off between
+/// speed and accuracy. Otherwise, calculating the size would take quite a long time.
+async fn prefix_storage_size(
+	client: &WsClient,
+	at: Option<Hash>,
+	prefix: StorageKey,
+	count: u32,
+) -> Option<(u64, usize)> {
+	match client.storage(prefix.clone(), at).await {
+		Ok(Some(d)) => return Some((d.0.len() as u64, 0)),
+		Ok(None) => (),
+		Err(e) => {
+			log::error!(target: LOG_TARGET, "Error while reading storage: {:?}", e);
+			return None;
+		},
+	}
+
+	let mut sum = 0;
+	let mut keys_count = 0;
+	let mut value_size = None;
+
+	let mut start_key = None;
+
+	loop {
+		let keys = client
+			.storage_keys_paged(Some(prefix.clone()), count, start_key, at)
+			.await
+			.unwrap_or(vec![]);
+
+		if keys.is_empty() {
+			break;
+		}
+
+		let current_keys = keys.len();
+		keys_count += current_keys;
+
+		for key in keys.clone() {
+			// don't really need to read for all, just for the first one to get the size.
+			if let Some(size) = value_size {
+				sum += size;
+			} else if let Ok(Some(value)) = client.storage(key.clone(), at).await {
+				value_size = Some(value.0.len() as u64);
+				sum += value.0.len() as u64;
+			}
+		}
+		if current_keys < count as usize {
+			break;
+		}
+
+		start_key = if let Some(last) = keys.last() { Some(last.clone()) } else { break };
+	}
+
+	Some((sum, keys_count))
+}
+
+#[tokio::main]
+async fn main() {
+	let now = std::time::Instant::now();
+
+	tracing_subscriber::fmt::try_init().unwrap();
+	let opt = Opt::from_args();
+
+	let rpc_client = substrate_rpc_client::ws_client(opt.uri).await.unwrap();
+	let mut modules: Vec<Pallet> = vec![];
+
+	// potentially replace head with the given hash
+	let head = ChainApiClient::<(), _, Header, ()>::finalized_head(&rpc_client).await.unwrap();
+
+	let at = opt.at.or(Some(head));
+
+	let runtime = rpc_client.runtime_version(at).await.unwrap();
+
+	println!("Scraping at block {:?} of {}({})", at, runtime.spec_name, runtime.spec_version,);
+
+	let raw_metadata = rpc_client.metadata(at).await.unwrap();
+	let prefixed_metadata = <RuntimeMetadataPrefixed as codec::Decode>::decode(&mut &*raw_metadata)
+		.expect("Runtime Metadata failed to decode");
+	let metadata = prefixed_metadata.1;
+
+	let mut total_size = 0;
+	match metadata {
+		RuntimeMetadata::V14(inner) => {
```

Comment:
We at least need to support Metadata V15 as well. 

Moreover, I wonder if this whole crate can now be done better with `subxt`, or `frame-metadata` crate?